runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8763,6 +8784,308 @@ TR::CompilationInfoPerThreadBase::performAOTLoad(
+ 
+      if(TR::Options::canJITCompile())      
+         {
+         TR::Options *options = TR::Options::getJITCmdLineOptions();
+
+         J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+         U_16 maxLength = J9UTF8_LENGTH(J9ROMMETHOD_SIGNATURE(romMethod));
+         char typeString[maxLength];
+         if(MJIT::nativeSignature(method, typeString))
+            {
+            return 0;
+            }
+
+         U_16 paramCount = MJIT::getParamCount(typeString, maxLength);
+         MJIT::ParamTableEntry paramTableEntries[paramCount];
+
+         int error_code = 0;
+         MJIT::RegisterStack stack = MJIT::mapIncomingParams(typeString, maxLength, &error_code, paramTableEntries, paramCount, logFileFP);
+         if(error_code)
+            MJIT_COMPILE_ERROR(0, method);
+      
+         MJIT::ParamTable paramTable(paramTableEntries, paramCount, &stack);
+
+#define MAX_BUFFER_SIZE 1024
+         // zeroed buffer for generated code
+         MJIT::CodeGenGC mjitCGGC(logFileFP);
+         MJIT::CodeGenerator mjit_cg(_jitConfig, vmThread, logFileFP, vm, &paramTable, compiler, &mjitCGGC);
+         char* buffer = (char*)mjit_cg.allocateCodeCache(MAX_BUFFER_SIZE, &vm, vmThread);
+         memset(buffer, 0, MAX_BUFFER_SIZE);
+         codeCache = mjit_cg.getCodeCache();
+
+         // provide enough space for CodeCacheMethodHeader
+         char* cursor = &buffer[sizeof(OMR::CodeCacheMethodHeader)];
+
+         buffer_size_t buffer_size = 0;
+
+         TR_PersistentJittedBodyInfo* bodyInfo;
+               
+         char *magicWordLocation, *first2BytesPatchLocation;
+         buffer_size_t code_size = mjit_cg.generatePrePrologue(
+            cursor,
+            method,
+            &magicWordLocation,
+            &first2BytesPatchLocation,
+            &bodyInfo);
+         MJIT_COMPILE_ERROR(code_size, method);
+            
+         compiler->cg()->setPrePrologueSize(code_size);
+         buffer_size += code_size;
+         MJIT_ASSERT(logFileFP, buffer_size < MAX_BUFFER_SIZE, "Buffer overflow after pre-prologue");
+
+         trfprintf(this->getCompilation()->getOutFile(), "\ngeneratePrePrologue\n");
+         for(int32_t i = 0; i < code_size; i++)
+            {
+            trfprintf(this->getCompilation()->getOutFile(), "%02x\n", ((unsigned char)cursor[i]) & (unsigned char)0xff);
+            }
+
+         cursor += code_size;
+
+         // (extra2) start point should point to prolog
+         char * prologue_address = cursor;
+
+         // generate debug breakpoint
+         if (comp()->getOption(TR_EntryBreakPoints))
+            {  
+            code_size = mjit_cg.generateDebugBreakpoint(cursor);
+            cursor += code_size;
+            buffer_size += code_size;
+            }
+
+         char * jitStackOverflowPatchLocation = NULL;
+
+         mjit_cg.setPeakStackSize(romMethod->maxStack * 8);
```

Comment:
What is `8` here? Does it assume a 64-bit target?

Reply:
We are starting with x86-64, however this should not have been a magic value and we do want to make supporting other architectures easier. I added a method for getting the pointer size from the MicroJIT code generator.